### DISCUSSION TITLE
v2.11.0

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#5863] [Releases & Announcements]
## Changelog

- Macros
- Ability to search in help centre public portals
- Support for location messages in the WhatsApp channel
- Support for custom attributes in automation
- Ability to paste files/images from the Clipboard 
- Realtime update for conversation attributes on agent dashboard
- Redis connection performance improvements
- Numerous other enhancements and bug fixes

Thanks to @smartdev58, @NitinRamnani, @KubesDavid,  @NusretOzates, @DanielGannage, @ChadBurggraf,  @jacsonsantospht, @givetimetolife, @singpolyma, @arkadiyayvazyan, @CristianDuta,   @mjattiot, @satoshinotdead, @salman652,    for the contributions

<hr /><em>This discussion was created from the release <a href='https://github.com/chatwoot/chatwoot/releases/tag/v2.11.0'>v2.11.0</a>.</em>

**Comment by hasanbasri1993** [comment]
Thanks u for update

**Comment by glauberssoutodossantos** [comment]
Hi @sojan-official. Tks for update.

How this works?
`_Support for custom attributes in automation_`


I did not see how to do that in chatwoot portal :(
![image](https://user-images.githubusercontent.com/99494469/202261646-6ceefaa0-37a6-46e3-a1b1-62f5e233bb45.png)

**Comment by sanjayws** [comment]
> Hi @sojan-official. Tks for update.
> 
> 
> 
> How this works?
> 
> `_Support for custom attributes in automation_`
> 
> 
> 
> 
> 
> I did not see how to do that in chatwoot portal :(
> 
> ![image](https://user-images.githubusercontent.com/99494469/202261646-6ceefaa0-37a6-46e3-a1b1-62f5e233bb45.png)
> 
> 

I think you may need to define a custom attribute first to use it. I have it working on my instance

**Reply by glauberssoutodossantos** [reply]
Hi @sanjayws it works. Thanks for helping.

**Comment by luisdanielhj** [comment]
Hi, I update some images to my self host chatwoot, how can I upgrade, when I try to do it the terminal shows me that I need to update manually how can I do it?

<img width="473" alt="Screenshot 2022-11-17 at 18 17 23" src="https://user-images.githubusercontent.com/106158566/202587598-7e7be87c-ece2-450b-a7c7-20061cf5c650.png">

**Reply by pranavrajs** [reply]
@luisdanielhj Unfortunately, we can't support custom changes in the installation, please use the standard package from Chatwoot.

**Reply by luisdanielhj** [reply]
these was the only changes that I made, if I restore to the stocks images Ill be able to upgrade it?

**Comment by sojan-official** [comment]
## 2.11.0 Dev Changelog

### API Endpoints

#### Additions

- `/api/v1/accounts/#{account.id}/macros/attach_file` [POST]
  -  https://github.com/chatwoot/chatwoot/pull/5609/files
  - description: Endpoint to upload file attachments required for macros
  
#### Deletions
- none
